source/core_http_client.c

```diff
@@ -1980,7 +1990,7 @@ static HTTPStatus_t receiveAndParseHttpResponse( const TransportInterface_t * pT
                                                  const HTTPRequestHeaders_t * pRequestHeaders )
 {
     HTTPStatus_t returnStatus = HTTPSuccess;
-    size_t totalReceived = 0U;
+    uint64_t totalReceived = 0U;
```

Comment:
size_t by definition must be able to contain the entirety of any object.

Reply:
I thought that the use of size_t was a MISRA violation, after checking this change doesn't cause a violation. Will swap this back to the size_t on next commit.